internal/agent/device/publisher/publisher.go

```diff
@@ -125,6 +128,19 @@ func (n *publisher) pollAndPublish(ctx context.Context) {
 		n.log.Debugf("Dialing management API took: %v", duration)
 	}
 	if err != nil {
+		// Check for device not found error - handle certificate wiping and restart
+		if errors.Is(err, client.ErrDeviceNotFound) {
+			n.log.Warn("Device not found on management server")
```

Comment:
question: is this an error I think this is very unexpected during normal operation